nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/QueryRecord.java

```diff
@@ -396,7 +396,7 @@ public void process(final OutputStream out) throws IOException {
             session.remove(createdFlowFiles);
             session.transfer(original, REL_FAILURE);
         } catch (final Exception e) {
-            getLogger().error("Unable to query {} due to {}", new Object[] {original, e});
+            getLogger().error("Unable to query {} due to {}", new Object[] {original, e.getCause() == null ? e : e.getCause()});
```

Comment:
I don't think we want to be getting the 'cause' for the general Exception - we should log the exception itself. We get the cause for SQLException only because we know that it's generally wrapping another exception.

Reply:
Yeah, this one was a strange one. Since the calcite cast function throwing the cannot handle Object exception was getting trapped here and the message was super vague so had to dig into the app logs to see the stack trace.

Reply:
Yeah unfortunately for this processor, when an Exception is thrown it's kind of difficult to figure out without the logs. But if we were to unwrap the cause, that could make even the logs confusing, which is even worse.